custom/icds/tests/test_aww_no_submissions.py

```diff
@@ -22,49 +20,92 @@ def setUpClass(cls):
+    def now(self):
+        return datetime.utcnow()
 
-    def test_form_sent_today(self, aww_user_ids):
-        aww_user_ids.return_value = {self.user.get_id: self.today}
+    def test_form_sent_today(self):
+        self.app_fact.last_form_submission_date = self.now
+        self.app_fact.save()
         messages = run_indicator_for_user(self.user, AWWSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 0)
 
-    def test_form_sent_seven_days_ago(self, aww_user_ids):
-        aww_user_ids.return_value = {self.user.get_id: self.today - timedelta(days=7)}
+    def test_form_sent_just_under_seven_days_ago(self):
+        self.app_fact.last_form_submission_date = self.now - timedelta(days=6, hours=23)
+        self.app_fact.save()
         messages = run_indicator_for_user(self.user, AWWSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 0)
 
-    def test_form_sent_eight_days_ago(self, aww_user_ids):
-        aww_user_ids.return_value = {self.user.get_id: self.today - timedelta(days=8)}
+    def test_form_sent_eight_days_ago(self):
+        self.app_fact.last_form_submission_date = self.now - timedelta(days=8)
+        self.app_fact.save()
         messages = run_indicator_for_user(self.user, AWWSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
-        self.assertTrue('one week' in messages[0])
+        self.assertIn('one week', messages[0])
 
-    def test_form_sent_thirty_days_ago(self, aww_user_ids):
-        aww_user_ids.return_value = {self.user.get_id: self.today - timedelta(days=30)}
+    def test_form_sent_thirty_days_ago(self):
+        self.app_fact.last_form_submission_date = self.now - timedelta(days=29, hours=23)
+        self.app_fact.save()
         messages = run_indicator_for_user(self.user, AWWSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
-        self.assertTrue('one week' in messages[0])
+        self.assertIn('one week', messages[0])
 
-    def test_form_sent_thirty_one_days_ago(self, aww_user_ids):
-        # last submissions only looks 30 days into past
-        aww_user_ids.return_value = {}
+    def test_form_sent_thirty_one_days_ago(self):
+        self.app_fact.last_form_submission_date = self.now - timedelta(days=31)
+        self.app_fact.save()
         messages = run_indicator_for_user(self.user, AWWSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
-        self.assertTrue('one month' in messages[0])
+        self.assertIn('one month', messages[0])
+
+    def test_no_last_form_submission(self):
+        self.app_fact.last_form_submission_date = None
+        self.app_fact.save()
+        messages = run_indicator_for_user(self.user, AWWSubmissionPerformanceIndicator, language_code='en')
+        self.assertEqual(len(messages), 1)
+        self.assertIn('one month', messages[0])
```

Comment:
could we also add a test using a user who has no entries in the app status fact at all? it would be a good test to have around for regression tests against any later changes

Reply:
https://github.com/dimagi/commcare-hq/pull/21581/commits/08a05d928f4e792803b7b330939b1dcf180d52ef